packages/sitecore-jss-cli/src/create/index.ts

```diff
@@ -3,73 +3,144 @@ import fs from 'fs';
 import glob from 'glob';
 import path from 'path';
 
+interface PackageJsonConfig {
+  appName: string;
+  graphQLEndpointPath?: string;
+  sitecoreDistPath?: string;
+}
+
+interface PackageJson {
+  name: string;
+  config: PackageJsonConfig;
+}
+
 /**
- * @param {string} projectFolder
- * @param {string} name
- * @param {string} hostName
+ * Escapes a string literal for use in RegEx
+ * @param {string} literal
  */
-export function applyNameToProject(projectFolder: string, name: string, hostName: string) {
-  console.log(chalk.cyan(`Setting package name ${name}...`));
+function escapeRegex(literal: string) {
+  return literal.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
+}
 
-  const packagePath = path.join(projectFolder, 'package.json');
+/**
+ * @param {PackageJson} pkg package.json object
+ * @param {string} name App name
+ * @param {string} [replaceName] Optional token which will enable "name replacement mode". If omitted, default/conventional values are used.
+ */
+export function applyNameToPackageJson(pkg: PackageJson, name: string, replaceName?: string) {
```

Comment:
Looking at this from the perspective of someone new to JSS, it's not obvious how we expect people to use "name replacement mode". Do we show example of valid `replaceName` values somewhere? I don't remember seeing this in docs. Maybe a backlog item for docs is all that's needed. I think adding an example of how the transform works to the function docs would be helpful too.

Reply:
@anastasiya29 Agree, this isn't the clearest. Struggled to find a balance between this and previous functionality to maintain backwards compatibility. The `replaceName` argument is new with this PR, so not documented anywhere - though I don't think the whole `jss-create.js` and `applyNameToProject` were ever documented that I can find. I'll add a backlog item to document this better.